presto-main/src/main/java/com/facebook/presto/server/InternalCommunicationConfig.java

```diff
@@ -27,6 +28,7 @@
     private boolean kerberosEnabled;
     private boolean kerberosUseCanonicalHostname = true;
     private boolean binaryTransportEnabled;
+    private DataSize taskUpdateSizeLimit = new DataSize(800, DataSize.Unit.MEGABYTE);
```

Comment:
I understand you want to set it to a high level to avoid current queries from failure. But 800MB is... clearly too large and is not a good default value.

Usually we set the default value to a more reasonable size (e.g. 16MB), and we change our production config files to make sure no query will fail. I can show you how we do it :)